### PR TITLE
MessageFormat parse panic

### PR DESCRIPTION
**Pull request by DarjaLeonova**

### CODE REVIEW

**Review by kaaposc — APPROVED** [review]

File: pkg/messageformat/lex.go [hunk 1]
```diff
@@ -211,12 +211,13 @@ func lexExpr(l *lexer) stateFn {
 		}
 	case '$':
 		return lexVariable(l)
+
 	case ':':
 		return lexFunction(l)
 	case '+':
-		return lexOpeningFunction(l)
+		return processFunction(l, v)
```
Comment (janishorsts) [inline pkg/messageformat/lex.go:218]: No need for `processFunction`, it makes it harder to read the code, and slower, too.

One option is ...

```suggestion
		if l.exprDepth > 1 {
			return lexOpeningFunction(l)
		}
		
		l.backup()
		
		return lexText(l)
```

File: pkg/messageformat/lex.go [hunk 2]
```diff
@@ -370,3 +371,18 @@ func isNameChar(v rune) bool {
...
+			return lexOpeningFunction(l)
+		case '-':
+			return lexClosingFunction(l)
+		}
+	}
+
+	l.backup()
+
+	return lexText(l)
+}
```
Comment (janishorsts) [inline pkg/messageformat/lex.go:388]: ... or pass the lex function

```suggestion
func processFunction(l *lexer, f ...) stateFn {
	if l.exprDepth > 1 {
		f(l)
	}

	l.backup()

	return lexText(l)
}
```

**Comment by DarjaLeonova** [comment]
@janishorsts looks like we have the same problem with variable ($) and function (:) chars. 
We know that opening and closing functions are always depthLevel > 1 . {(1)expression block {(2)+button}{-button}(2) }(1)
but variables can be written as "match {$count :number} (depth level == 1)" ... 
So the issue will be when the text block contains $ or :.
Should i fix it in the same PR or create new one ?

**Comment by janishorsts** [comment]
@DarjaLeonova please fix it in this PR, the change is small. No need for micromanagement.

**Review by VladislavsPerkanuks — APPROVED** [review]
Seems good

**Review by janishorsts — APPROVED** [review]

**Pull request merged**